File: Tests/ApiTests/Qwen3_6_35b_a3bTests.swift
import Testing
import Foundation
@testable import Api

struct Qwen3_6_35b_a3bTests {
    @Test func fundedUserReturnsReply() async throws {
        let result = await Api.qwen3_6_35b_a3b(
            user: testUser,
            password: testPassword,
            messages: [(role: .user, content: "say hi in one word")]
        )
        #expect(result.count == 2)
        #expect(result.last?.role == .assistant)
        let reply = result.last?.content ?? ""
        #expect(reply != "Could not respond")
        #expect(!reply.isEmpty)
    }

    @Test func cancellationReturnsFallback() async throws {
        let task = Task {
            await Api.qwen3_6_35b_a3b(
                user: testUser,
                password: testPassword,
                messages: [(role: .user, content: "write a long story")]
            )
        }
        try await Task.sleep(nanoseconds: 100_000_000)
        task.cancel()
        let start = Date()
        let result = await task.value
        let elapsed = Date().timeIntervalSince(start)
        #expect(result.last?.content == "Could not respond")
        #expect(elapsed < 1.0, "cancel should resolve in <1s, took \(elapsed)s")
    }
}

File: Rust/tests/qwen3_6_35b_a3b_tests.rs
#![cfg(not(target_arch = "wasm32"))]

#[path = "api_tests.rs"] mod api_tests;
use api_tests::*;
use rust_ffi::rust_ffi_qwen3_6_35b_a3b;

#[test]
fn funded_user_returns_reply() {
    let u = cs(&test_user()); let pw = cs(TEST_PASSWORD);
    let m = cs(r#"[{"role":"User","content":"say hi in one word"}]"#);
    let mut len = 0usize;
    let r = rust_ffi_qwen3_6_35b_a3b(u.as_ptr(), pw.as_ptr(), m.as_ptr(), std::ptr::null(), &mut len);
    let bytes = unsafe { take(r, len) };
    let arr = serde_json::from_slice::<serde_json::Value>(&bytes).unwrap();
    let arr = arr.as_array().unwrap();
    assert_eq!(arr.len(), 2);
    assert_eq!(arr.last().unwrap()["role"], "Assistant");
    let reply = arr.last().unwrap()["content"].as_str().unwrap();
    assert_ne!(reply, "Could not respond");
    assert!(!reply.is_empty());
}

#[test]
fn empty_messages_returns_fallback() {
    let u = cs(&test_user()); let pw = cs(TEST_PASSWORD);
    let m = cs("[]");
    let mut len = 0usize;
    let r = rust_ffi_qwen3_6_35b_a3b(u.as_ptr(), pw.as_ptr(), m.as_ptr(), std::ptr::null(), &mut len);
    let bytes = unsafe { take(r, len) };
    let arr = serde_json::from_slice::<serde_json::Value>(&bytes).unwrap();
    let arr = arr.as_array().unwrap();
    assert_eq!(arr.len(), 1);
    assert_eq!(arr.last().unwrap()["content"], "Could not respond");
}
